pytest_mergify/quarantine.py

```diff
@@ -0,0 +1,101 @@
+import dataclasses
+import requests
+import pytest
+import _pytest.nodes
+import typing
+from pytest_mergify import utils
+
+
+@dataclasses.dataclass
+class MergifyTestsQuarantine:
+    token: typing.Optional[str]
+    repo_name: typing.Optional[str]
+    api_url: str
+    quarantined_tests: typing.List[str] = dataclasses.field(
+        init=False, default_factory=list
+    )
+    init_warning_logs: typing.List[str] = dataclasses.field(
+        init=False, default_factory=list
+    )
+
+    def __post_init__(self) -> None:
+        if not utils.is_in_ci():
+            return
+
+        if self.token is None:
+            self.init_warning_logs.append(
+                "A Mergify API token is required to use CI Insights Quarantine"
+            )
+            return
+
+        if self.repo_name is None:
+            self.init_warning_logs.append(
+                "No valid repository name found, skipping CI Insights Quarantine setup"
+            )
+            return
+
+        # Query all the quarantined tests
+        try:
+            owner, repository = self.repo_name.split("/")
+        except ValueError:
+            self.init_warning_logs.append(
+                f"Repository name '{self.repo_name}' has an unexpected format (expected 'owner/repository'), skipping CI Insights Quarantine setup"
+            )
+            return
+
+        branch_name = utils.get_branch_name_for_quarantine()
+        if branch_name is None:
+            self.init_warning_logs.append(
+                "No valid branch name found, skipping CI Insights Quarantine setup",
+            )
+            return
+
+        url = f"{self.api_url}/v1/ci/{owner}/repositories/{repository}/quarantines"
+
+        try:
+            quarantine_resp: requests.Response = requests.get(
+                url,
+                headers={"Authorization": f"Bearer {self.token}"},
+                params={"branch": branch_name},
+                timeout=10,
+            )
+        except requests.ConnectionError as exc:
+            self.init_warning_logs.append(
+                f"Failed to connect to Mergify's API, tests won't be quarantined. Error: {str(exc)}"
+            )
+            return
+
+        if quarantine_resp.status_code == 402:
+            # No CI Insights Quarantine subscription, skip it.
+            return
+
+        try:
+            quarantine_resp.raise_for_status()
+        except requests.HTTPError as exc:
+            self.init_warning_logs.append(
+                f"Error when querying Mergify's API, tests won't be quarantined. Error: {str(exc)}"
+            )
+            return
+
+        self.quarantined_tests = quarantine_resp.json()["quarantined_tests"]
+
+    def is_test_quarantined(self, test_item: _pytest.nodes.Item) -> bool:
```

Comment:
adding:
```suggestion
    def __contains__(self, test_item: _pytest.nodes.Item) -> bool:
```
would make it sexier ;)